webbpsf/detectors.py

```diff
@@ -0,0 +1,329 @@
+        out_ipc_ppc = convolve(out_ipc, kernel)
+
+        psf_hdulist[ext].header['IPCINST'] = ('NIRCam', 'Interpixel capacitance (IPC)')
+        psf_hdulist[ext].header['IPCTYPA'] = (det2sca[det], 'Post-Pixel Coupling (PPC)')
+        psf_hdulist[ext].data = out_ipc_ppc
+
+    if inst =='MIRI':
+        a = webbpsf.constants.INSTRUMENT_IPC_DEFAULT_KERNEL_PARAMETERS[inst]
+        webbpsf.webbpsf_core._log.info("Detector IPC: MIRI")
+        alpha = webbpsf.constants.INSTRUMENT_IPC_DEFAULT_KERNEL_PARAMETERS[inst][0]
+        beta = webbpsf.constants.INSTRUMENT_IPC_DEFAULT_KERNEL_PARAMETERS[inst][1]
+        c = webbpsf.constants.INSTRUMENT_IPC_DEFAULT_KERNEL_PARAMETERS[inst][2]  # real observation noise adjustment
+        miri_kernel = np.array([[c, beta, c],
+                                [alpha, 1 - 2 * alpha - 2 * beta - 4 * c, alpha],
+                                [c, beta, c]])
+        kernel = CustomKernel(miri_kernel)
+        # apply to DET_DIST
+        ext = 'DET_DIST'
+        out  = convolve(psf_hdulist[ext].data, kernel)
+        psf_hdulist[ext].header['IPCINST'] = ('MIRI', 'Interpixel capacitance (IPC)')
+        psf_hdulist[ext].header['IPCTYPA'] = (alpha, 'coupling coefficient alpha')
+        psf_hdulist[ext].header['IPCTYPB'] = (beta, 'coupling coefficient beta')
+        psf_hdulist[ext].data = out
+
+    #
+    if inst == 'NIRISS':
+        ext = 'DET_DIST'
+        # this set-up the input variables as required by Kevin Volk IPC code
+        image = psf_hdulist[ext].data
+        xposition = psf_hdulist[ext].header["DET_X"]
+        yposition = psf_hdulist[ext].header["DET_Y"]
+        # find the voidmask fits file
+        voidmask10 = os.path.join(utils.get_webbpsf_data_path(), 'NIRISS', 'IPC', 'voidmask10.fits')
+
+        if os.path.exists(voidmask10):
+            maskimage = fits.getdata(voidmask10)
+        else:
+            maskimage = None
+            webbpsf.webbpsf_core._log.info("Error reading the file voidmask10.fits.  Will assume a non-void position.")
+
+        nchannel = int(yposition) // 512
+        try:
+            flag = maskimage[nchannel, int(xposition)]
+        except:
+            # This marks the pixel as non-void by default if the maskimage is not
+            # read in properly
+            flag = 0
+        frag1 = ['A', 'B', 'C', 'D']
+        frag2 = ['notvoid', 'void']
+
+        ipcname = 'ipc5by5median_amp' + frag1[nchannel] + '_' + \
+                      frag2[flag] + '.fits'
+        ipc_file = os.path.join(utils.get_webbpsf_data_path(), 'NIRISS', 'IPC', ipcname)
+        if os.path.exists(ipc_file):
+            ipckernel = fits.getdata(ipc_file)
+            newimage = signal.fftconvolve(image, ipckernel, mode='same')
+            psf_hdulist[ext].header['IPCINST'] = ('NIRISS', 'Interpixel capacitance (IPC)')
+            psf_hdulist[ext].header['IPCTYPA'] = (ipcname, 'kernel file used for IPC correction')
+        else:
+            newimage = numpy.copy(image)
```

Comment:
```suggestion
            newimage = np.copy(image)
```

Reply:
small bug that breaks one of your examples in the pr
(NIRISS PSF with IPC effects)

Reply:
Fixed just now, thanks.